app/routes/_layout+/bookings.update-existing.tsx

```diff
@@ -0,0 +1,431 @@
+      title: "Bookings",
+    };
+
+    const modelName = {
+      singular: "booking",
+      plural: "bookings",
+    };
+
+    const totalPages = Math.ceil(bookingCount / perPage);
+
+    const hints = getClientHint(request);
+
+    const items = bookings.map((b) => {
+      if (b.from && b.to) {
+        const from = new Date(b.from);
+        const displayFrom = getDateTimeFormat(request, {
+          dateStyle: "short",
+          timeStyle: "short",
+        }).format(from);
+
+        const to = new Date(b.to);
+        const displayTo = getDateTimeFormat(request, {
+          dateStyle: "short",
+          timeStyle: "short",
+        }).format(to);
+
+        return {
+          ...b,
+          displayFrom: displayFrom.split(","),
+          displayTo: displayTo.split(","),
+          metadata: {
+            ...b,
+            displayFrom: displayFrom.split(","),
+            displayTo: displayTo.split(","),
+          },
+        };
+      }
+      return b;
+    });
+    return json(
+      data({
+        showModal: true,
+        header,
+        bookings: items,
+        search,
+        page,
+        bookingCount: bookingCount,
+        totalPages,
+        perPage,
+        modelName,
+        ids: ids.length ? ids : undefined,
+        indexType,
+        hints,
+      }),
+      {
+        headers: [
+          setCookie(await setSelectedOrganizationIdCookie(organizationId)),
+        ],
+      }
+    );
+  } catch (cause) {
+    const reason = makeShelfError(cause, { userId });
+    throw json(error(reason), { status: reason.status });
+  }
+}
+
+const processBooking = async (
+  indexType: string,
+  bookingId: string,
+  assetIds: string[] | undefined,
+  kitIds: string[] | undefined
+) => {
+  try {
+    let finalAssetIds: string[] = [];
+    let booking;
+    if (indexType === "kits" && kitIds && kitIds.length > 0) {
+      const promises = [
+        getAvailableKitAssetForBooking(kitIds),
+        getExistingBookingDetails(bookingId),
+      ];
+      const [assetIdsFromKits, bookingDetails] = await Promise.all(promises);
+
+      finalAssetIds = assetIdsFromKits as string[];
+      booking = bookingDetails;
+    } else if (indexType === "assets" && assetIds && assetIds.length > 0) {
+      const promises = [
+        getAvailableAssetsForBooking(assetIds),
+        getExistingBookingDetails(bookingId),
+      ];
+
+      const [assets, bookingDetails] = await Promise.all(promises);
+      finalAssetIds = assets as string[];
+      booking = bookingDetails;
+    } else {
+      throw new ShelfError({
+        cause: null,
+        message: "Invalid operation. Please contact support.",
+        label: "Booking",
+      });
+    }
+
+    if (finalAssetIds.length === 0) {
+      throw new ShelfError({
+        cause: null,
+        message: "No assets available.",
+        label: "Booking",
+      });
+    }
+
+    return {
+      finalAssetIds,
+      bookingInfo: booking,
+    };
+  } catch (err) {
```

Comment:
Use `cause` please. Also use the message from the cause if it exists. That way the user knows what actually happened.